src/axolotl/utils/chat_templates.py

```diff
@@ -17,20 +44,67 @@ def chat_templates(user_choice: str):
-    }
-
-    if user_choice in templates:
-        return templates[user_choice]
+        user_choice = user_choice[
+            len(_DEFAULT_FALLBACK_CHATML_TEMPLATE_CHOICE_PREFIX) :
+        ]
+        LOG.warning(
+            f"No chat template found on tokenizer, falling back to {user_choice}. It is recommended to set --train_on_inputs to True for the model to learn this chat template."
```

Comment:
Is this warning accurate? I wasn't aware of it.

Reply:
This is assuming if chat template is missing then most likely the model was not trained to chat at all

If the model was indeed trained for chat but chat template is not present in tokenizer config then this warning would be incorrect - but this should be a rare case

Reply:
This makes sense after reading the code in more detail. It's for the condition if setting template with fallback.